netman/core/objects/bond.py

```diff
@@ -11,16 +11,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import warnings
 
+from netman.core.objects.interface import BaseInterface
 
-class Bond(object):
-    def __init__(self, number=None, interface=None, link_speed=None, members=None):
+
+class Bond(BaseInterface):
+    def __init__(self, number=None, link_speed=None, members=None, **interface):
+        super(Bond, self).__init__(**interface)
         self.number = number
-        self.interface = interface
         self.link_speed = link_speed
         self.members = members or []
 
-    def __eq__(self, other):
-        if isinstance(other, type(self)):
-            return self.__dict__ == other.__dict__
-        return False
+    @property
+    def interface(self):
```

Comment:
This is not tested anymore, i suggest adding a backward compatibility test in (or beside) the juniper test to ensure that when doing get_bonds, the object .interface.* still resolve to the right thing, and interface.name must be "aeX"